client/platform/desktop/backend/serializers/viame.ts

```diff
@@ -0,0 +1,267 @@
+  const array = [...str.matchAll(regexp)];
+  if (array.length) {
+    return array[1].slice(1);
+  }
+  return null;
+}
+
+function _rowInfo(row: string[]) {
+  if (row.length < 9) {
+    throw new Error('malformed row: too few columns');
+  }
+  if (row[0].match(CommentRegex) !== null) {
+    throw new Error('malformed row: comment row');
+  }
+  return {
+    trackId: parseInt(row[0], 10),
+    filename: row[1],
+    frame: parseInt(row[2], 10),
+    bounds: (
+      row.slice(3, 7).map((v) => Math.round(parseFloat(v))) as
+      [number, number, number, number]
+    ),
+    fishLength: parseFloat(row[8]),
+  };
+}
+
+function _deduceType(value: string): boolean | number | string {
+  if (value === 'true') {
+    return true;
+  }
+  if (value === 'false') {
+    return false;
+  }
+  try {
+    return parseFloat(value);
+  } catch {
+    return value;
+  }
+}
+
+/**
+ * Simplified from python variant.  Does not handle duplicate type/key pairs
+ * within a single feature.
+ *
+ * @param type geojson feature type
+ * @param coords 2D float array
+ * @param key optional feature key string
+ */
+function _createGeoJsonFeature(
+  type: 'Point' | 'LineString' | 'Polygon',
+  coords: number[][],
+  key = '',
+) {
+  const geoFeature: GeoJSON.Feature<TrackSupportedFeature, GeoJSON.GeoJsonProperties> = {
+    type: 'Feature',
+    properties: { key },
+    geometry: {
+      type,
+      coordinates: [],
+    },
+  };
+  if (type === 'Polygon') {
+    geoFeature.geometry.coordinates = [coords];
+  } else if (type === 'Point') {
+    [geoFeature.geometry.coordinates] = coords;
+  } else {
+    geoFeature.geometry.coordinates = coords;
+  }
+  return geoFeature;
+}
+
+function _parseRow(row: string[]) {
+  // Create empty feature collection
+  const geoFeatureCollection:
+    GeoJSON.FeatureCollection<TrackSupportedFeature, GeoJSON.GeoJsonProperties> = {
+      type: 'FeatureCollection',
+      features: [],
+    };
+  let attributes: StringKeyObject | undefined;
+  const trackAttributes: StringKeyObject = {};
+  const cpStarti = 9; // Confidence pairs start at i=9
+  const confidencePairs: ConfidencePair[] = row
+    .slice(cpStarti, row.length)
+    .map((value, j) => {
+      if (j % 2 !== 0) {
+        // Filter out ODDs
+        return ['', 0] as ConfidencePair;
+      }
+      const i = j + cpStarti;
+      if ((i + 1) < row.length && row[i] && row[i + 1] && !row[i].startsWith('(')) {
+        return [row[i], parseFloat(row[i + 1])] as ConfidencePair;
+      }
+      return ['', 0] as ConfidencePair;
+    })
+    .filter((val) => val[0] !== '');
+  const headTail: [number, number][] = [];
+  const start = 9 + (confidencePairs.length * 2);
+  row.slice(start).forEach((value) => {
+    /* Head */
+    const head = getCaptureGroups(HeadRegex, value);
+    if (head !== null) {
+      headTail[0] = [parseFloat(head[0]), parseFloat(head[1])];
+      geoFeatureCollection.features.push(_createGeoJsonFeature('Point', [headTail[0]], 'head'));
```

Comment:
If you do the changes above for array[0] instead of array[1].splice(1) you need to modify these indices.  If not you would need to modify the attribute indicies.